runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingPlugin.java

```diff
@@ -263,46 +259,50 @@ public void onGameTick(GameTick event)
 			}
 		}
 
-		final LocalPoint cameraPoint = new LocalPoint(client.getCameraX(), client.getCameraY());
-
-		final NPCQuery query = new NPCQuery()
-			.idEquals(Ints.toArray(FishingSpot.getSPOTS().keySet()));
-		NPC[] spots = queryRunner.runQuery(query);
-		// -1 to make closer things draw last (on top of farther things)
-		Arrays.sort(spots, Comparator.comparing(npc -> -1 * npc.getLocalLocation().distanceTo(cameraPoint)));
-		fishingSpots = spots;
+		inverseSortSpotDistanceFromPlayer();
 
-		// process minnows
-		for (NPC npc : spots)
+		for (NPC npc : fishingSpots)
 		{
-			FishingSpot spot = FishingSpot.getSPOTS().get(npc.getId());
-
+			final FishingSpot spot = FishingSpot.getSPOTS().get(npc.getId());
 			if (spot == null)
 			{
-				continue;
+				return;
 			}
 
 			if (spot == FishingSpot.MINNOW && config.showMinnowOverlay())
 			{
-				int id = npc.getIndex();
-				MinnowSpot minnowSpot = minnowSpots.get(id);
+				final int id = npc.getIndex();
+				final MinnowSpot minnowSpot = minnowSpots.get(id);
+
 				// create the minnow spot if it doesn't already exist
-				if (minnowSpot == null)
-				{
-					minnowSpots.put(id, new MinnowSpot(npc.getWorldLocation(), Instant.now()));
-				}
-				// if moved, reset
-				else if (!minnowSpot.getLoc().equals(npc.getWorldLocation()))
+				// or if it was moved, reset it
+				if (minnowSpot == null
+					|| !minnowSpot.getLoc().equals(npc.getWorldLocation()))
 				{
 					minnowSpots.put(id, new MinnowSpot(npc.getWorldLocation(), Instant.now()));
 				}
 			}
 		}
 	}
 
+	@Subscribe
+	public void onNpcSpawned(NpcSpawned event)
+	{
+		final NPC npc = event.getNpc();
+
+		if (!FishingSpot.getSPOTS().containsKey(npc.getId()))
+		{
+			return;
+		}
+
+		fishingSpots.add(npc);
+		inverseSortSpotDistanceFromPlayer();
+	}
+
 	@Subscribe
 	public void onNpcDespawned(NpcDespawned npcDespawned)
 	{
+
```

Comment:
remove this empty line

Reply:
I think this was meant to be me removing the npc from the fishingSpots array, but I forgot it somehow... Good catch.